client/src/hooks/hooks.js

```diff
@@ -0,0 +1,41 @@
+import { locationSearch } from '../api/weatherApi';
+import { useState, useEffect } from 'react';
+import { timeoutToSearchCities } from '../globalConsts';
+import { startLengthToSearchCitisThroughApi } from '../globalConsts';
+
+export const useCitiesSearch = initialState => {
+  const [searchText, setSearchText] = useState(initialState);
+  const [matchingCities, setMatchingCities] = useState([]);
+  const [needToShowCitiesWindow, setNeedToShowCitiesWindow] = useState(false);
```

Comment:
maybe something like isCitiesWindowShown will be better. 
isSomething for boolean values.